### PR TITLE
	fix agent rpc panic bug

### PR DESCRIPTION
**Pull request by gpt1111**

### CODE REVIEW

File: src/agent/deployer/deployer.go [hunk 1]
```diff
@@ -122,8 +122,9 @@ func (s *Deployer) StartModuleDeployLoop() error {
 			return nil
 		}
 		if err != nil {
-			log.Errorf("Failed to read stream from DeplyModule(), error: %v", err)
-			return err
+			retry.ExpBackOffWithLimit(s.ConnectToAPIServer)
+			log.Errorf("Failed to read stream from DeplyModule(), reconnecting, error: %v", err)
+			continue //return
```
Comment (nascentcore-eng) [inline src/agent/deployer/deployer.go:127]: This is not needed, as the outer layer already has retry, see
communicateWithNode at 
https://github.com/tricorder-observability/Starship/blob/7df9f0e5d32cf10e7bd4b53ec22da1552fb9a4c3/src/agent/cmd/main.go#L76

And the for loop that calls this:
 https://github.com/tricorder-observability/Starship/blob/7df9f0e5d32cf10e7bd4b53ec22da1552fb9a4c3/src/agent/cmd/main.go#L64

File: src/agent/deployer/deployer.go [hunk 2]
```diff
@@ -148,8 +149,8 @@ func (s *Deployer) StartModuleDeployLoop() error {
 		err = s.sendResp(resp)
 		// TODO(yzhao): Need to handle error correctly, the code below ignores EoF error.
 		if grpcerr.IsUnavailable(err) {
-			log.Errorf("Streaming connection with api-server is broken, error: %v", err)
-			return err
+			log.Errorf("Streaming connection with api-server is broken, error: %v", err)	
+			continue //return err
```
Comment (nascentcore-eng) [inline src/agent/deployer/deployer.go:153]: Why use continue?

File: src/agent/proc-info/collector.go [hunk 3]
```diff
@@ -60,39 +62,56 @@ func NewCollector(hostSysRootPath, apiServerAddr, nodeName string) *Collector {
...
 
+	return err
+}
+
+func (c *Collector) connect() error {
+	grpcConn, err := grpc.Dial(c.apiServerAddr,
+		grpc.WithTransportCredentials(insecure.NewCredentials()))
+	if err != nil {
+		return fmt.Errorf("failed to connect to API server at '%s', error: %v", c.apiServerAddr, err)
+	}
```
Comment (nascentcore-eng) [inline src/agent/proc-info/collector.go:94]: You can follow what's done in agent's deployer 

https://github.com/tricorder-observability/Starship/blob/7df9f0e5d32cf10e7bd4b53ec22da1552fb9a4c3/src/agent/cmd/main.go#L64

https://github.com/tricorder-observability/Starship/blob/7df9f0e5d32cf10e7bd4b53ec22da1552fb9a4c3/src/agent/cmd/main.go#L76

And do the similar thing, i.e., repeating the connection in outer scope code, not inside the handling loop

**Review by nascentcore-eng — CHANGES_REQUESTED** [review]

File: src/agent/deployer/deployer.go (hunk 1 above)
Comment (gpt1111) [reply src/agent/deployer/deployer.go:127]: good

**Pull request closed without merging**